tests/testthat/test-PackageDependencyReporter.R

```diff
@@ -0,0 +1,147 @@
+context("Package Dependency Reporter Tests")
+
+##### TEST SET UP #####
+
+rm(list = ls())
+# Configure logger (suppress all logs in testing)
+loggerOptions <- futile.logger::logger.options()
+if (!identical(loggerOptions, list())){
+  origLogThreshold <- loggerOptions[[1]][['threshold']]
+} else {
+  origLogThreshold <- futile.logger::INFO
+}
+futile.logger::flog.threshold(0)
+
+##### TESTS #####
+
+## Structure Available ##
+
+test_that('PackageDependencyReporter structure is as expected', {
+  
+  expect_named(object = PackageDependencyReporter$public_methods
+               , expected = c(
+                 "calculate_metrics", 
+                 "extract_network", 
+                 "clone"
+               )
+               , info = "Available public methods for PackageDependencyReporter not as expected."
+  )
+  
+  expect_named(object = PackageDependencyReporter$public_fields
+               , expected = NULL
+               , info = "Available public fields for PackageDependencyReporter not as expected."
+  )
+  
+  expect_named(object = PackageDependencyReporter$private_methods
+               , expected = NULL
+               , info = "Available private methods for PackageDependencyReporter not as expected."
+  )
+  
+  expect_named(object = PackageDependencyReporter$private_fields
+               , expected = NULL
+               , info = "Available private fields for PackageDependencyReporter not as expected."
+  )
+  
+})
+
+### USAGE OF PUBLIC AND PRIVATE METHODS AND FIELDS
+
+test_that('PackageDependencyReporter Methods Work', {
```

Comment:
this is a beautiful way to test R6 classes. Really dig the way you set this up. Nice work!